contracts/RFQ.sol

```diff
@@ -123,15 +136,33 @@ contract RFQ is IRFQ, StrategyBase, ReentrancyGuard, SignatureValidator, BaseLib
     }
 
     // settle
-    function _settle(RFQLibEIP712.Order memory _order, GroupedVars memory _vars) internal returns (uint256) {
+    function _settle(
+        RFQLibEIP712.Order memory _order,
+        SpenderLibEIP712.SpendWithPermit memory _makerAssetPermit,
+        SpenderLibEIP712.SpendWithPermit memory _takerAssetPermit,
+        GroupedVars memory _vars,
+        bytes memory _makerAssetPermitSig,
+        bytes memory _takerAssetPermitSig
+    ) internal returns (uint256) {
         // Transfer taker asset to maker
         if (address(weth) == _order.takerAssetAddr) {
             // Deposit to WETH if taker asset is ETH
             require(msg.value == _order.takerAssetAmount, "RFQ: insufficient ETH");
             weth.deposit{ value: msg.value }();
             weth.transfer(_order.makerAddr, _order.takerAssetAmount);
         } else {
-            spender.spendFromUserTo(_order.takerAddr, _order.takerAssetAddr, _order.makerAddr, _order.takerAssetAmount);
+            require(
```

Comment:
I think it may be necessary to consider the different strategy contract.
For example, like L2Deposit using deposit struct,
and AMMWrapper and LimitOrder using their own order struct.

I think it might consider the different strategy contracts.
For example, like L2Deposit uses the deposit struct,
And AMMWrapper and LimitOrder use their own order struct.